File: C#/Squeaky Clean.cs
using System;
using System.Text;


public static class Identifier
{
    public static string Clean(string identifier)
    {
        if (string.IsNullOrEmpty(identifier))
        {
            return identifier;
        }
        
        var dashIsPresent = false;

        StringBuilder sb = new StringBuilder();
        foreach (char c in identifier)
        {

                if (' '.Equals(c))
                {
                    sb.Append("_");
                }
                else if (char.IsControl(c))
                {
                    sb.Append("CTRL");
                }
                else if ('-'.Equals(c))
                { 
                    dashIsPresent = true;
                }
                else if (dashIsPresent == true)
                { 
                    sb.Append(char.ToUpperInvariant(c));
                    dashIsPresent = false;
                }
                else if (char.IsLetter(c) & !(c >= 'α' && c <= 'ω'))
                { 
                    sb.Append(c);
                }
            }
        
    string cleaned = sb.ToString();

    return cleaned;

    }
}

File: Java/Squeaky_Clean.java
class SqueakyClean
{
    static String clean(String identifier)
    {
        StringBuilder returnString = new StringBuilder();
        int i;

        for (i=0; i < identifier.length(); i++)
        {
            Character text = identifier.charAt(i);
            if(Character.isWhitespace(text))
            {
                returnString.append('_');
                continue;
            }
            else if (Character.isISOControl(text))
            {
                returnString.append("CTRL");
                continue;
            }
            else if (text == '-')
            {
                returnString.append(identifier.toUpperCase().charAt(i+1));
                i = i + 1;
                continue;
            }
            else if(!Character.isLetter(text))
            {
                continue;
            }
            else if(text >= 'α' && text <= 'ω')
            {
                continue;
            }
            else
            {
                returnString.append(text);
            }
        }
        return returnString.toString();
    }
}